src/main/java/uk/gov/pay/publicauth/app/config/XRayDataSourceFactory.java

```diff
@@ -0,0 +1,21 @@
+package uk.gov.pay.publicauth.app.config;
+
+import com.codahale.metrics.MetricRegistry;
+import io.dropwizard.db.DataSourceFactory;
+import io.dropwizard.db.ManagedDataSource;
+import io.dropwizard.db.ManagedPooledDataSource;
+import io.dropwizard.db.PooledDataSourceFactory;
+
+
+/**
+ * Created by markjones on 20/07/2017.
+ */
+public class XRayDataSourceFactory extends DataSourceFactory implements PooledDataSourceFactory {
+
+    @Override
+    public ManagedDataSource build(MetricRegistry metricRegistry, String s) {
+        ManagedDataSource managedPooledDataSource = super.build(metricRegistry, s);
+       // ((ManagedPooledDataSource) managedPooledDataSource).setJdbcInterceptors("com.amazonaws.xray.sql.postgres.TracingInterceptor;");
```

Comment:
This is commented out as the liquibase migrations were breaking without the AWS XRAY default behaviour of erroring when there are no in progress segments. Comment this back in and set the env var AWS_XRAY_CONTEXT_MISSING to LOG_ERROR to avoid it blowing up and getting DB monitoring back @brid